### PR TITLE
GHA workflow migration

### PR DESCRIPTION
**Pull request by skripted-io**
### Resolves

This PR adds a GHA workflow, migrated from CCI.

### Test Coverage

I couldn't get the tests to complete using ACT but luckily they worked in the action GH runner.

~~Please note that also this workflow calls "npm publish --tag $NPM_TAG --dry-run". Once approved please remove the "--dry-run" part to test the actual deploy. Tnx!~~ fixed with a condition.

### CODE REVIEW

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                  NPM version: $(npm --version)
+                  GitHub ref: ${{ github.ref }}
+                  GitHub head ref: ${{ github.head_ref }}
+                  EOF
+
+            - name: Install & update dependencies
+              run:
+                  | # By default, --production=false is implied, so this flag isn't strictly necessary.
+                  npm install
+                  npm update
```
Comment (cwillisf) [inline .github/workflows/ci-cd.yml]: Oof, I didn't realize this repo still used npm install! I would fully transition to npm ci instead:
```suggestion
            - name: Install dependencies
              run: npm ci
```

File: .github/workflows/ci-cd.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+            - name: Install Playwright browsers
+              run: npx playwright install-deps
+
+            - name: Setup & Test
+              run: |
+                  npm ci
+                  mkdir -p ./test/results
+                  npm test
```
Comment (cwillisf) [inline .github/workflows/ci-cd.yml]: If you accept the `npm ci` change above, it won't be necessary here:
```suggestion
            - name: Setup & Test
              run: |
                  mkdir -p ./test/results
                  npm test
```

File: .github/workflows/ci-cd.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                  echo "Deploying version $RELEASE_VERSION to $NPM_TAG"
+                  npm set //registry.npmjs.org/:_authToken=$NPM_TOKEN
+                  npm publish --tag $NPM_TAG --dry-run
+
+            - name: Check Release Version and Create Tag
+              run: |
+                  if npm info | grep -q $RELEASE_VERSION; then 
+                    git tag $RELEASE_VERSION 
+                    git push origin $RELEASE_VERSION
+                  fi
```
Comment (cwillisf) [inline .github/workflows/ci-cd.yml:87]: Thanks for making this easier to read!

**Review by cwillisf — CHANGES_REQUESTED** [review]
I'd recommend updating the NPM dependency installation steps, but other than that this looks good!

**Comment by skripted-io** [comment]
@cwillisf I didn't bother creating a new PR but added a Env based on what I saw in Ron's script and think is useful. Also noticed we aren't using a "main" but "master" branch. And apparently linting was still commented out.

**Review by cwillisf — APPROVED** [review]
Looks reasonable. Thanks!

**Comment by cwillisf** [comment]
For reference, the commitlint thing accepts these categories:
`[build, chore, ci, docs, feat, fix, perf, refactor, revert, style, test]`
For these GHA things, you can probably "cheat" and use "ci: did a thing" for all of them unless you're specifically altering tests or docs or something :)

**Pull request merged**